### PR TITLE
Prepare the app for deployment on Heroku

### PR DESCRIPTION
**Pull request by gclssvglx**
Following the advice to [Set up Heroku review apps for pull requests](https://docs.publishing.service.gov.uk/manual/review-apps.html) and [Test & build a project with GitHub Actions](https://docs.publishing.service.gov.uk/manual/test-and-build-a-project-with-github-actions.html) - this change:

* Updates rubocop to use the `rubocop-govuk` gem - and corrects issues found 
* Adds a `Procfile`
* Updates the `Gemfile` with advice on keeping the Ruby version consistent 
* Adds an `app.json` file - this defines the project structure and any required Heroku add-ons
* Adds GitHub Actions to act as a CI layer with builds and tests the application
* Updates the production URL to `DATABASE_URL` for Heroku

[Trello](https://trello.com/c/gTnbKiF1/673-move-the-foreign-travel-deep-link-checker-application-to-alphagov-govuks-heroku)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,82 +1,40 @@
...
-
-# Use Sass to process CSS
-# gem "sassc-rails"
-
-# Use Active Storage variants [https://guides.rubyonrails.org/active_storage_overview.html#transforming-images]
-# gem "image_processing", "~> 1.2"
-
-gem 'httparty'
-gem 'nokogiri'
+ruby "2.7.2"
```
Comment (kevindew) [inline Gemfile]: We tend to use: https://github.com/alphagov/smart-answers/blob/b2c4c330dbe128ade02600ebec5b0c3239ae9bca/Gemfile#L3 to save duplicating the .ruby-version file.

File: Gemfile [hunk 2]
```diff
@@ -1,82 +1,40 @@
...
+
+gem "bootsnap", require: false
+gem "httparty"
+gem "importmap-rails"
+gem "jbuilder"
+gem "nokogiri"
+gem "pg", "~> 1.1"
+gem "puma", "~> 5.0"
+gem "rails", "~> 7.0.2", ">= 7.0.2.2"
+gem "redis", "~> 4.0"
```
Comment (kevindew) [inline Gemfile]: Minor given the status of this repo, but these lines don't match GOV.UK conventions: https://docs.publishing.service.gov.uk/manual/conventions-for-rails-applications.html#gemfile-organisation

File: Procfile [hunk 3]
```diff
@@ -0,0 +1 @@
+web: bin/rails server -p $PORT -e $RAILS_ENV
```
Comment (kevindew) [inline Procfile:1]: Minor, the commit message for this is spelt wrong s/Profile/Procfile/

File: Gemfile [hunk 4]
```diff
@@ -3,7 +3,7 @@
 source "https://rubygems.org"
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby "2.7.2"
+ruby File.read(".ruby-version").chomp
```
Comment (kevindew) [inline Gemfile:6]: Nice 👍

File: app.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    "PLEK_SERVICE_CONTENT_STORE_URI": {
+      "value": "https://www.gov.uk/api"
+    },
+    "PLEK_SERVICE_SEARCH_URI": {
+      "value": "https://www.gov.uk/api"
+    },
+    "PLEK_SERVICE_STATIC_URI": {
+      "value": "assets.digital.cabinet-office.gov.uk"
+    },
```
Comment (kevindew) [inline app.json]: Do you know if you need all these env vars? they seem the type that are needed for GOV.UK dependencies that I doubt this app has.

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+
+      - name: Setup the database
+        env:
+          RAILS_ENV: test
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: foreign_travel_content_checker_test
```
Comment (kevindew) [inline .github/workflows/ci.yml]: It'd be more common to set this up through a DATABASE_URL environment variable. 

Typically for testing apps we've used TEST_DATABASE_URL e.g: https://github.com/alphagov/govuk-docker/blob/main/projects/content-publisher/docker-compose.yml#L31-L33

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: foreign_travel_content_checker_test
```
Comment (kevindew) [inline .github/workflows/ci.yml]: I'm not sure you actually need much of this as per: https://docs.publishing.service.gov.uk/manual/test-and-build-a-project-with-github-actions.html#gov-uk-rails-application-with-postgres-redis-yarn-and-gov-uk-content-schemas-dependencies

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: Setup the database
+        env:
+          RAILS_ENV: test
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: foreign_travel_content_checker_test
+        run: |
+          bundle exec rails db:migrate
+          bundle exec rails db:seed
```
Comment (kevindew) [inline .github/workflows/ci.yml]: You'll probably want this to create the database too. You can do `db:reset` which will drop, create and seed.

It's not recommended to using db:migrate as a way to set-up a database. I've not found the authoritative documentation of this but this one at least sets the tone: https://edgeguides.rubyonrails.org/active_record_migrations.html#old-migrations

File: config/database.yml [hunk 9]
```diff
@@ -58,6 +58,9 @@ development:
 test:
   <<: *default
   database: foreign_travel_content_checker_test
+  host: localhost
+  username: postgres
+  password: postgres
```
Comment (kevindew) [inline config/database.yml]: You should be able to remove this

File: lib/tasks/security.rake [hunk 10]
```diff
@@ -2,5 +2,5 @@
 
 desc "Run Brakeman"
 task security: [:environment] do
-  sh "bundle exec brakeman"
+  sh "bundle exec brakeman . --except CheckRenderInline --no-summary --quiet"
```
Comment (kevindew) [inline lib/tasks/security.rake]: Might want to put this in a separate commit and explain why these options exist.

**Review by kevindew — COMMENTED** [review]
Lots of great additions here 👍  I've added a few comments to resolve.

File: Gemfile (hunk 1 above)
Comment (gclssvglx) [reply Gemfile]: Thanks - have added this in this commit https://github.com/alphagov/foreign-travel-content-checker/pull/1/commits/6261f07bf6fd4a5f9617a4f56e31311f656638b3

File: Procfile (hunk 3 above)
Comment (gclssvglx) [reply Procfile:1]: Thanks - fixed

File: app.json (hunk 5 above)
Comment (gclssvglx) [reply app.json]: Nope, the app doesn't need any of these. Just a copy-and-paste issue as I wasn't sure. Fixed.

File: .github/workflows/ci.yml (hunk 8 above)
Comment (gclssvglx) [reply .github/workflows/ci.yml]: Thanks - fixed

File: Gemfile (hunk 2 above)
Comment (gclssvglx) [reply Gemfile]: Thanks - fixed

File: .github/workflows/ci.yml (hunk 6 above)
Comment (gclssvglx) [reply .github/workflows/ci.yml]: Thanks - removed

File: .github/workflows/ci.yml (hunk 7 above)
Comment (gclssvglx) [reply .github/workflows/ci.yml]: Thanks - removed

File: config/database.yml (hunk 9 above)
Comment (gclssvglx) [reply config/database.yml:63]: Thanks - removed

File: .github/workflows/ci.yml (hunk 6 above)
Comment (gclssvglx) [reply .github/workflows/ci.yml]: When I tried to remove these, I got a `postgres service is unhealthy` error. Found this advice: https://github.com/jveillet/tentacles-dashboard/issues/205, then this https://github.community/t/mysql-service-never-comes-up-healthy-in-action/16890 which suggests that something has changed with GitHub actions and DRYing it no longer works 🤷

File: .github/workflows/ci.yml (hunk 7 above)
Comment (gclssvglx) [reply .github/workflows/ci.yml]: See https://github.com/alphagov/foreign-travel-content-checker/pull/1#discussion_r826119433

File: config/database.yml (hunk 9 above)
Comment (gclssvglx) [reply config/database.yml]: See https://github.com/alphagov/foreign-travel-content-checker/pull/1#discussion_r826119433

**Review by kevindew — APPROVED** [review]
